### ISSUE TITLE
[v2 FEATURE]: Layer-dependent hidden size for the feedforward neural network

### ISSUE THREAD

**Issue by akshatzalte** [chemprop/chemprop#1361]
## Is your feature request related to a problem? Please describe.

Following the MLPDS spring 2026 discussion, there is a request to allow each layer of the readout FFN to have its own hidden size, configurable from the CLI. Currently `--ffn-hidden-dim` takes a single int that is applied uniformly to every hidden layer, so users cannot try non-rectangular profiles (funnel, hourglass, etc.) without dropping into the Python API.
 
## Use-cases / examples
 
- **Funnel / decreasing widths** (e.g. `[600, 300, 150]`) — common pattern for tapered readout heads.
- **Hourglass / bottleneck** (e.g. `[512, 64, 512]`) — representational bottleneck as a regularizer for small datasets.
- **Reproducing literature baselines** that use non-uniform readout MLPs.
- **Architecture exploration** comparing uniform vs tapered FFNs at matched parameter count.
## Desired solution / workflow
 
Extend `--ffn-hidden-dim` to accept `nargs="+"`, preserving the single-int form for backward compatibility:
 
```bash
# Legacy (unchanged): uniform stack of width 300
chemprop train ... --ffn-num-layers 3 --ffn-hidden-dim 300
 
# New: explicit per-layer profile
chemprop train ... --ffn-hidden-dim 600 300 150
```
 
Sketch:
 
```python
parser.add_argument("--ffn-hidden-dim", type=int, nargs="+", default=[DEFAULT_HIDDEN_DIM])
 
# in build_model:
hidden_dims = args.ffn_hidden_dim
if len(hidden_dims) == 1:
    hidden_dims = hidden_dims * args.ffn_num_layers          # legacy path
else:
    if args.ffn_num_layers not in (None, len(hidden_dims)):
        raise ArgumentError("--ffn-num-layers must match len(--ffn-hidden-dim)")
    args.ffn_num_layers = len(hidden_dims)
```
 
The FFN builder in `chemprop/nn/ffn.py` would accept `hidden_dims: Sequence[int]` and iterate when stacking `nn.Linear` layers.

**Comment by JacksonBurns** [comment]
The majority of this change will happen in our `MLP` class here:

https://github.com/chemprop/chemprop/blob/34656ef22e8091401e3d3334e12558d871d5b427/chemprop/nn/ffn.py#L24-L67

All of the actual predictor FFNs will just their type hints updated.
